AtReconstruction/AtPatternModification/AtBraggCurveFinder.cxx

```diff
@@ -0,0 +1,162 @@
+#include "AtBraggCurveFinder.h"
+
+#include "AtFindVertex.h"
+#include "AtPattern.h"
+
+#include <TH1F.h>
+
+using XYZPoint = ROOT::Math::XYZPoint;
+
+ClassImp(AtBraggCurveFinder);
+
+void AtBraggCurveFinder::InitializePSA()
+{
+   fPSA = std::make_unique<AtPSAHitPerTB>();
+   fPSA->SetReplaceTraceIntegral(false);
+   fPSA->Init();
+}
+
+void AtBraggCurveFinder::Init()
+{
+   if (fNeedPSA)
+      InitializePSA();
+
+   uint64_t timeSeed = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+   std::seed_seq ss{uint32_t(timeSeed & 0xffffffff), uint32_t(timeSeed >> 32)};
+   fRNG.seed(ss);
+}
+
+void AtBraggCurveFinder::ModifyPatternEvent()
+{
+   std::vector<AtTrack> &tracks = fPatternEvent->GetTrackCand();
+   LOG(info) << "Applying AtBraggCurveFinder modification to " << tracks.size() << " tracks.";
+
+   if (fRawEvent == nullptr)
+      LOG(warning) << "No AtRawEvent was passed to the At3DBraggFinder, so it will not be used. The resulting bragg "
+                      "curve may have more fluctuations in the ELoss per bin than expected!";
+
+   for (auto &track : tracks)
+      ProcessTrack(track);
+}
```

Comment:
This is then just the base class implementation so doesn't need to be overwritten in your case.